Documentation/Evolution/RegexSyntax.md

```diff
@@ -0,0 +1,245 @@
+# Regular Expression Syntax
+
+- Authors: Hamish Knight, Michael Ilseman
+
+## Introduction
+
+We aim to parse a superset of the syntax accepted by a variety of popular regular expression engines.
+
+**TODO: Elaborate**
+
+## Engines supported
+
+We aim to implement a syntactic superset of:
+
+- [PCRE 2][pcre2-syntax], an "industry standard" of sorts, and a rough superset of Perl, Python, etc.
+- [Oniguruma][oniguruma-syntax], an internationalization-oriented engine with some modern features
+- [ICU][icu-syntax], used by NSRegularExpression, a Unicode-focused engine.
+- [.NET][.net-syntax]'s regular expressions, which support delimiter-balancing and some interesting minor details on conditional patterns.
+- **TODO: List Java here? It doesn't really add any more syntax than the above other than `\p{javaLowerCase}`**
+
+We also intend to achieve at least Level 1 (**TODO: do we want to promise Level 2?**) [UTS#18][uts18] conformance, which specifies regular expression matching semantics without mandating any particular syntax. However we can infer syntactic feature sets from its guidance.
+
+## Regex syntax supported
+
+### General syntax
+
+The following syntax are supported by all the above engines.
+
+#### Alternation
+
+```
+Regex       -> '' | Alternation
+Alternation -> Concatenation ('|' Concatenation)*
+```
+
+This is the operator with the lowest precedence in a regular expression, and checks if any of its branches match the input.
```

Comment:
I'm really glad to see a grammar. Prose could be more of the form of briefly describing what it is/does. Something in-between the terseness of pcre2syntax and the fuller specification of pcre2pattern.

Maybe something like "`|` denotes what is formally called an alternation, or a choice between alternatives. Any number of alternatives may appear, including empty alternatives. This operator has the lowest precedence of all operators in a regex literal."